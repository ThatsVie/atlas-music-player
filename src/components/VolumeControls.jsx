import React from 'react';
import { Volume2 } from 'lucide-react';

const VolumeControls = () => {
  return (
    <div className="flex items-center space-x-4 mt-6">
      <Volume2 className="w-7 h-7 text-brightPink" />
      <input
        type="range"
        min="0"
        max="100"
        defaultValue="50"
        className="w-full h-2 rounded-full bg-magenta accent-teal focus:outline-none focus:ring-2 focus:ring-brightPink transition"
        aria-label="Volume Control"
      />
    </div>
  );
};

export default VolumeControls;
