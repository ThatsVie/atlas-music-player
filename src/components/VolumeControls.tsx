import React from 'react';
import { Volume2 } from 'lucide-react';

interface VolumeControlsProps {
  volume: number;
  onVolumeChange: (newVolume: number) => void;
}

const VolumeControls: React.FC<VolumeControlsProps> = ({
  volume,
  onVolumeChange,
}) => {
  // changes to the volume slider
  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const newVolume = parseInt(event.target.value, 10);
    onVolumeChange(newVolume);
  };

  return (
    <div className='flex items-center space-x-4 mt-6'>
      <label htmlFor='volume-control' className='sr-only'>
        Volume Control
      </label>
      {/* Volume Icon */}
      <Volume2 className='w-7 h-7 text-darkMagenta dark:text-lightTeal' />
      {/* Volume Slider */}
      <input
        id='volume-control'
        type='range'
        min='0'
        max='100'
        value={volume}
        onChange={handleChange}
        className='w-full h-2 rounded-full bg-lightTeal dark:bg-darkerTeal accent-teal dark:accent-lightTeal focus:outline-none focus:ring-2 focus:ring-brightPink dark:focus:ring-lightTeal transition'
      />
    </div>
  );
};

export default VolumeControls;
