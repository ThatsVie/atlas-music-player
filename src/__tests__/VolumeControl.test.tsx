import { render, fireEvent, screen } from '@testing-library/react';
import { describe, test, expect, vi, afterEach } from 'vitest';
import VolumeControls from '@/components/VolumeControls';

describe('VolumeControls Component', () => {
  const mockOnVolumeChange = vi.fn();

  afterEach(() => {
    vi.clearAllMocks();
  });

  test('renders correctly with initial volume', () => {
    // Arrange & Act
    const { container } = render(
      <VolumeControls volume={50} onVolumeChange={mockOnVolumeChange} />
    );

    // Assert
    expect(container).toMatchSnapshot();
  });

  test('displays correct initial volume value', () => {
    // Arrange
    render(<VolumeControls volume={75} onVolumeChange={mockOnVolumeChange} />);

    // Act
    const inputElement = screen.getByLabelText('Volume Control');

    // Assert
    expect(inputElement.getAttribute('value')).toBe('75');
  });

  test('calls onVolumeChange with the correct value when volume changes', () => {
    // Arrange
    render(<VolumeControls volume={30} onVolumeChange={mockOnVolumeChange} />);

    // Act
    const inputElement = screen.getByLabelText('Volume Control');
    fireEvent.change(inputElement, { target: { value: '60' } });

    // Assert
    expect(mockOnVolumeChange).toHaveBeenCalledTimes(1);
    expect(mockOnVolumeChange).toHaveBeenCalledWith(60);
  });

  test('handles keyboard events (ArrowLeft and ArrowRight) without propagating', () => {
    // Arrange
    const stopPropagationMock = vi.fn();
    render(<VolumeControls volume={40} onVolumeChange={mockOnVolumeChange} />);

    const inputElement = screen.getByLabelText('Volume Control');
    const event = new KeyboardEvent('keydown', { key: 'ArrowRight', bubbles: true });
    Object.defineProperty(event, 'stopPropagation', { value: stopPropagationMock });

    // Act
    inputElement.dispatchEvent(event);

    // Assert
    expect(stopPropagationMock).toHaveBeenCalledTimes(1);
  });
});
