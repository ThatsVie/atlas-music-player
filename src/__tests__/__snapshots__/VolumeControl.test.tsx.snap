// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VolumeControls Component > renders correctly with initial volume 1`] = `
<div>
  <div
    class="flex items-center space-x-4 mt-6"
  >
    <label
      class="sr-only"
      for="volume-control"
    >
      Volume Control
    </label>
    <svg
      class="lucide lucide-volume2 w-7 h-7 text-darkMagenta dark:text-lightTeal"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11 4.702a.705.705 0 0 0-1.203-.498L6.413 7.587A1.4 1.4 0 0 1 5.416 8H3a1 1 0 0 0-1 1v6a1 1 0 0 0 1 1h2.416a1.4 1.4 0 0 1 .997.413l3.383 3.384A.705.705 0 0 0 11 19.298z"
      />
      <path
        d="M16 9a5 5 0 0 1 0 6"
      />
      <path
        d="M19.364 18.364a9 9 0 0 0 0-12.728"
      />
    </svg>
    <input
      class="w-full h-2 rounded-full bg-lightTeal dark:bg-darkerTeal accent-teal dark:accent-lightTeal focus:outline-none focus:ring-2 focus:ring-brightPink dark:focus:ring-lightTeal transition"
      id="volume-control"
      max="100"
      min="0"
      type="range"
      value="50"
    />
  </div>
</div>
`;
